src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs

```diff
@@ -33,8 +33,9 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var managementClient = ManagementClientConnections.GetOrAdd(ConnectionKey, _ => CreateManagementClient());
-                _ = await managementClient.GetQueueRuntimePropertiesAsync(_queueName, cancellationToken);
+                var client = ClientConnections.GetOrAdd(ConnectionKey, _ => CreateClient());
+                var receiver = client.CreateReceiver(_queueName);

```

Comment:
`CreateReceiver` returns something that doesn't implement `IDisposable`. Right?

Reply:
Good catch. Looks like it actually implements IAsyncDisposable and is recommended to be cached.

From the comments on `ServiceBusReceiver`:

>  /// The <see cref="ServiceBusReceiver" /> is safe to cache and use for the lifetime of an
    /// application or until the <see cref="ServiceBusClient" /> that it was created by is disposed.
    /// Caching the receiver is recommended when the application is consuming messages
    /// regularly or semi-regularly.

I'll refactor to be caching that in addition to the ServiceBusClient